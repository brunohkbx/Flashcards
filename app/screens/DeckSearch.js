import React, { Component } from 'react';
import { EvilIcons } from '@expo/vector-icons';
import { View, KeyboardAvoidingView } from 'react-native';
import { Paragraph } from 'react-native-paper';
import SearchToolbar from '../components/SearchToolbar';

class DeckSearch extends Component {
  static navigationOptions = ({ navigation }) => {
    return {
      header: <SearchToolbar
                handleChangeText={navigation.getParam('handleChangeText')}
                value={navigation.getParam('query')}
                handleIconPress={() => navigation.goBack()}
              />
    };
  };

  state = { query: '' }

  componentDidMount() {
    const { navigation }  = this.props;

    navigation.setParams({ handleChangeText: this.onChangeText });
  }

  onChangeText = query => {
    const { navigation } = this.props;

    this.setState({ query });
    navigation.setParams({ query });
  }

  render() {
    return (
      <KeyboardAvoidingView
        style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}
        behavior="padding"
      >
        <View style={{alignItems: 'center'}}>
          <EvilIcons name="search" size={125} />
          <Paragraph>Enter a title and find your favorite deck</Paragraph>
        </View>
      </KeyboardAvoidingView>
    );
  }
}

export default DeckSearch;
