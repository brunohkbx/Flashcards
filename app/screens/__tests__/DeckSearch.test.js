import React from 'react';
import { shallow } from 'enzyme';
import DeckSearch from '../DeckSearch';

describe('DeckSearch', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      navigation: {
        setParams: jest.fn()
      }
    }, propOverrides)

    const wrapper = shallow(<DeckSearch {...defaultProps} />);

    return {
      wrapper
    }
  }

  it('renders properly', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  })
});
