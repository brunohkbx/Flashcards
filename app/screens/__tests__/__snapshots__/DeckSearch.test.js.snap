// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckSearch renders properly 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Icon
      allowFontScaling={false}
      name="search"
      size={125}
    />
    <Paragraph>
      Enter a title and find your favorite deck
    </Paragraph>
  </Component>
</KeyboardAvoidingView>
`;
